File: client/java/src/avt/OnSplashScr.java
package avt;

import java.io.IOException;
import javax.microedition.lcdui.Graphics;
import javax.microedition.lcdui.Image;
import main.Canvas;
import main.GameMidlet;

public final class OnSplashScr extends MyScreen {
   public static OnSplashScr me;
   public int splashScrStat = 0;
   public static Image imgBg;
   public static boolean isOpen = false;

   public static OnSplashScr gI() {
      return me == null ? (me = new OnSplashScr()) : me;
   }

   public final void switchToMe() {
      Canvas.listInfoSV.removeAllElements();
      Canvas.transTab = 0;
      Canvas.instance.setSize();
      OnScreen.isOngame = true;

      try {
         imgBg = Image.createImage(T.getPath() + "/on/logo.on");
      } catch (IOException var2) {
         var2.printStackTrace();
      }

      super.switchToMe();
   }

   public final void update() {
      if (this.splashScrStat > 21) {
         LoadMap.xDichChuyen_ = GameMidlet.avatar.x;
         LoadMap.C = GameMidlet.avatar.y;
         OnScreen.gI().switchToMe();
      } else if (this.splashScrStat == 0) {
         Canvas.paint.initResourceOne();
      }

      ++this.splashScrStat;
   }

   public final void paint(Graphics var1) {
      Canvas.paint.paintDefaultBg(var1);
      if (this.splashScrStat > 1) {
         var1.drawImage(imgBg, Canvas.hw, Canvas.hCan / 2, 3);
      }

      Canvas.paintPlus(var1);
   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/OnSplashScr.cs
using UnityEngine;

public class OnSplashScr : MyScreen
{
	public static OnSplashScr me;

	private static int runningPercent;

	public int splashScrStat;

	public static Image imgLogomainMenu;

	public static Image imgBg;

	public static bool isOpen;

	public static OnSplashScr gI()
	{
		return (me != null) ? me : (me = new OnSplashScr());
	}

	public override void switchToMe()
	{
		if (Screen.orientation == ScreenOrientation.Portrait)
		{
			Screen.orientation = ScreenOrientation.LandscapeLeft;
			Canvas.isRotateTop = 2;
		}
		Canvas.listInfoSV.removeAllElements();
		Canvas.transTab = 0;
		Canvas.instance.setSize((int)ScaleGUI.WIDTH, (int)ScaleGUI.HEIGHT);
		onMainMenu.isOngame = true;
		imgLogomainMenu = Image.createImagePNG(T.mode[AvMain.hd - 1] + "/hd/on/logo");
		if (imgBg == null)
		{
			imgBg = Image.createImage("backgroundOn");
		}
		base.switchToMe();
		Canvas.paint.clearImgAvatar();
	}

	public override void update()
	{
		if (!isOpen)
		{
			if (splashScrStat > 10)
			{
				onMainMenu.gI().switchToMe();
			}
			else if (splashScrStat == 1)
			{
				onMainMenu.initSize();
				Canvas.paint.initOngame();
			}
			splashScrStat++;
		}
	}

	public override void paint(MyGraphics g)
	{
		Canvas.paint.paintDefaultBg(g);
		g.drawImage(imgLogomainMenu, Canvas.hw, Canvas.hh, 3);
		Canvas.paintPlus(g);
	}
}
